cmd/crc/cmd/oc_env.go

```diff
@@ -1,65 +1,39 @@
 package cmd
 
 import (
-	"os"
-	"text/template"
-
 	"github.com/code-ready/crc/pkg/crc/constants"
 	"github.com/code-ready/crc/pkg/crc/errors"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/output"
 	"github.com/code-ready/crc/pkg/os/shell"
 	"github.com/spf13/cobra"
 )
 
-const (
-	ocEnvTmpl = `{{ .Prefix }}PATH{{ .Delimiter }}{{ .OcDirPath }}{{ .PathSuffix }}{{ .UsageHint }}`
-)
-
 var (
 	forceShell string
 )
 
-type OcShellConfig struct {
-	shell.ShellConfig
-	OcDirPath string
-	UsageHint string
-}
-
-func getOcShellConfig(ocPath string, forcedShell string) (*OcShellConfig, error) {
-	userShell, err := shell.GetShell(forcedShell)
-	if err != nil {
-		return nil, errors.Newf("Error running the oc-env command: %s", err.Error())
-	}
-
-	cmdLine := "crc oc-env"
-
-	shellCfg := &OcShellConfig{
-		OcDirPath: ocPath,
-	}
-
-	shellCfg.UsageHint = shell.GenerateUsageHint(userShell, cmdLine)
-	shellCfg.Prefix, shellCfg.Delimiter, shellCfg.Suffix, shellCfg.PathSuffix = shell.GetPrefixSuffixDelimiterForSet(userShell)
-
-	return shellCfg, nil
-}
-
-func executeOcTemplateStdout(shellCfg *OcShellConfig) error {
-	tmpl := template.Must(template.New("envConfig").Parse(ocEnvTmpl))
-	return tmpl.Execute(os.Stdout, shellCfg)
-}
-
 var ocEnvCmd = &cobra.Command{
 	Use:   "oc-env",
 	Short: "Add the 'oc' binary to PATH",
 	Long:  `Add the OpenShift client binary 'oc' to PATH`,
-	// This is required to make sure root command Persistent PreRun not run.
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {},
 	Run: func(cmd *cobra.Command, args []string) {
-		var shellCfg *OcShellConfig
-		shellCfg, err := getOcShellConfig(constants.CrcBinDir, forceShell)
+		userShell, err := shell.GetShell(forceShell)
+		if err != nil {
+			errors.ExitWithMessage(1, "Error running the oc-env command: %s", err.Error())
+		}
+
+		proxyConfig, err := machine.GetProxyConfig(constants.DefaultName)
 		if err != nil {
 			errors.Exit(1)
 		}
-		_ = executeOcTemplateStdout(shellCfg)
+		output.Outln(shell.GetPathEnvString(userShell, constants.CrcBinDir))
```

Comment:
@cfergeau once this is in then user is not able to execute `oc-env` if the VM is not present, which is not the case right now. This look like the change from user side, cc @gbraad ? should we just ignore the error if the machine not present?

```
$ ./crc oc-env
export PATH="/home/prkumar/.crc/bin:$PATH"
# Run this command to configure your shell:
# eval $(crc oc-env)

--- After this PR ---
$ ./crc oc-env
ERRO Machine "crc" does not exist. Use "crc start" to add a new one.
```

Reply:
In this case we should use an `echo` to force the printout.